src/pudl/output/ferc1.py

```diff
@@ -970,11 +978,41 @@ def create_exploded_table_assets() -> list[AssetsDefinition]:
 class MetadataExploder:
     """Combine a set of inter-related, nested table's metadata."""
 
-    def __init__(self, table_names: list[str]):
+    def __init__(
+        self,
+        table_names: list[str],
+        clean_xbrl_metadata_json: dict,
+        calculation_components_xbrl_ferc1: pd.DataFrame,
+    ):
         """Instantiate MetadataExploder."""
         self.table_names = table_names
+        self.calculation_components_xbrl_ferc1 = calculation_components_xbrl_ferc1
+        self.clean_xbrl_metadata_json = clean_xbrl_metadata_json
 
-    def boom(self, clean_xbrl_metadata_json: dict):
+    @property
+    def calculation_components_explosion(self):
+        """Reduce the calculation components to only the tables in the explosion.
+
+        First we restrict the calculation records to only those that are components of
+        xbrl_factoids within the set of tables that we are currently exploding. Then
+        we also have to remove the full set of calculaiton components that are a part of
+        calculations that have any components that are from tables outside of the set of
+        tables in this explosion.
+        """
+        calc_explode = self.calculation_components_xbrl_ferc1
+        calc_explode = calc_explode[calc_explode.table_name.isin(self.table_names)]
+        not_in_explosion_xbrl_factoids = list(
+            calc_explode.loc[
+                ~calc_explode.table_name_calculation_component.isin(self.table_names),
+                "xbrl_factoid",
+            ].unique()
+        )
+        calc_explode = calc_explode[
+            ~calc_explode.xbrl_factoid.isin(not_in_explosion_xbrl_factoids)
+        ].copy()
+        return calc_explode
+
+    def boom(self):
```

Comment:
should this just be:
```
@property
def table_metadata_explosion(self)
```

or some other such name like `xbrl_factiod_metadata`. now that this class has two care outputs: the table-factoid level metadata and the `calculation_components_explosion` this no longer seems like the right name

Reply:
i like that much better! because of the class name could it be `MetadataExploder.metadata` (or `MetadataExploder.xbrl_factoids`) & `MetadataExploder.calculations`? or something i think having exploded many times in the chain is a lil duplicative.

Reply:
I think `.metadata` and `.calculations` in context seems good.  We can assign them to some other variable name elsewhere when we read them if we need to transmit the 💥 context.